specs/gloas/beacon-chain.md

```diff
@@ -740,79 +831,74 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
+##### New `get_builder_withdrawals`
 
 ```python
-def is_builder_payment_withdrawable(
-    state: BeaconState, withdrawal: BuilderPendingWithdrawal
-) -> bool:
-    """
-    Check if a builder payment is withdrawable.
-    """
-    builder = state.validators[withdrawal.builder_index]
-    current_epoch = compute_epoch_at_slot(state.slot)
-    return not builder.slashed or current_epoch >= builder.withdrawable_epoch
-```
+def get_builder_withdrawals(
+    state: BeaconState,
+    withdrawal_index: WithdrawalIndex,
+    epoch: Epoch,
+    prior_withdrawals: Sequence[Withdrawal],
+) -> Tuple[Sequence[Withdrawal], WithdrawalIndex, uint64]:
+    withdrawals_limit = MAX_WITHDRAWALS_PER_PAYLOAD
 
-##### New `get_builder_withdrawable_balance`
+    processed_count: uint64 = 0
+    withdrawals: List[Withdrawal] = []
+    for withdrawal in state.builder_pending_withdrawals:
+        all_withdrawals = prior_withdrawals + withdrawals
+        has_reached_limit = len(all_withdrawals) == withdrawals_limit
+        if has_reached_limit:
+            break
 
-```python
-def get_builder_withdrawable_balance(builder: Validator, balance: Gwei) -> Gwei:
-    """
-    Get the withdrawable balance for a builder payment.
-    """
-    if builder.slashed:
-        return balance
-    elif balance > MIN_ACTIVATION_BALANCE:
-        return balance - MIN_ACTIVATION_BALANCE
-    else:
-        return Gwei(0)
+        builder_index = withdrawal.builder_index
+        withdrawals.append(
+            Withdrawal(
+                index=withdrawal_index,
+                validator_index=convert_builder_index_to_validator_index(builder_index),
+                address=withdrawal.fee_recipient,
+                amount=withdrawal.amount,
+            )
+        )
+        withdrawal_index += WithdrawalIndex(1)
+        processed_count += 1
+
+    return withdrawals, withdrawal_index, processed_count
 ```
 
-##### New `get_builder_withdrawals`
+##### New `get_builders_sweep_withdrawals`
 
 ```python
-def get_builder_withdrawals(
+def get_builders_sweep_withdrawals(
     state: BeaconState,
     withdrawal_index: WithdrawalIndex,
     epoch: Epoch,
     prior_withdrawals: Sequence[Withdrawal],
 ) -> Tuple[Sequence[Withdrawal], WithdrawalIndex, uint64]:
-    withdrawals_limit = MAX_WITHDRAWALS_PER_PAYLOAD - 1
+    builders_limit = min(len(state.builders), MAX_BUILDERS_PER_WITHDRAWALS_SWEEP)
+    withdrawals_limit = MAX_WITHDRAWALS_PER_PAYLOAD
 
     processed_count: uint64 = 0
     withdrawals: List[Withdrawal] = []
-    for withdrawal in state.builder_pending_withdrawals:
+    builder_index = state.next_withdrawal_builder_index
+    for _ in range(builders_limit):
         all_withdrawals = prior_withdrawals + withdrawals
-        is_withdrawable = withdrawal.withdrawable_epoch <= epoch
         has_reached_limit = len(all_withdrawals) == withdrawals_limit
-        if not is_withdrawable or has_reached_limit:
+        if has_reached_limit:
```

Comment:
This seems overly verbose, `has_reached_limit` is never used again in this function
```
if len(all_withdrawals) >= withdrawals_limit:
```
is super readable. Also, same as above, it doesn't make sense to pass the `prior` list, but even if passed, it doesn't make sense to now use `if len(prior_withdrawals) + len(withdrawals) >= withdrawals_limit` instead of concatenating lists. This will affect tests on the spec package.